serialisation/yaml/ActionContext.go

```diff
@@ -25,6 +25,25 @@ func (instance ContextBuilder) SetList(key string, value []map[string]interface{
 	return instance
 }
 
+//SetDefaults ...
+func (instance ContextBuilder) SetDefault(actionType string, key string, value interface{}) ContextBuilder {
+	var defaults map[string]interface{}
+	var actionDefaults map[string]interface{}
+
+	if instance.data["defaults"] == nil {
+		defaults = map[string]interface{}{}
```

Comment:
Ok I agree and I have changed a load of these now to type `core.ExecutionContext` which they should have been really.